homeassistant/components/konnected/config_flow.py

```diff
@@ -0,0 +1,732 @@
+                return self.async_abort(reason="not_konn_panel")
+
+            # extract host/port from ssdp_location
+            netloc = urlparse(discovery_info["ssdp_location"]).netloc.split(":")
+            self.host = self.context["host"] = netloc[0]
+            self.port = netloc[1]
+            self.model = discovery_info[ATTR_UPNP_MODEL_NAME]
+
+        except KeyError:
+            _LOGGER.error("Malformed Konnected SSDP info")
+            return self.async_abort(reason="unknown")
+
+        if any(
+            self.host == flow["context"].get("host")
+            for flow in self._async_in_progress()
+        ):
+            return self.async_abort(reason="already_in_progress")
+
+        if self.host in configured_hosts(self.hass):
+            return self.async_abort(reason="already_configured")
+
+        # brief delay to allow processing of recent status req
+        await asyncio.sleep(0.1)
+
+        # try to obtain the mac address from the device
+        try:
+            self.device_id = self.context["device_id"] = (
+                (await get_status(self.hass, self.host, self.port))
+                .get("mac")
+                .replace(":", "")
+            )
+
+        except CannotConnect:
+            return self.async_abort(reason="cannot_connect")
+
+        if not self.device_id:
+            return self.async_abort(reason="cannot_connect")
+
+        # look for a partially configured instance of this device and hijack it
+        for flow in self.hass.config_entries.flow.async_progress():
+            if flow["handler"] == self.handler and flow["flow_id"] != self.flow_id:
+                if (
+                    self.device_id == flow["context"].get("device_id")
+                    and self.cached_config
+                ):
+                    # steal the config data and abort other flow
+                    # only would happen if the other flow was imported config.yaml w/o host info
+                    _LOGGER.info("Konnected partially imported - completing cfg entry")
+                    config = copy.deepcopy(self.cached_config)
+                    config[CONF_HOST] = self.host
+                    config[CONF_PORT] = self.port
+                    self.hass.config_entries.flow.async_abort(flow["flow_id"])
+                    return await self.async_create_or_update_entry(config)
+
+        # if this device exists but the host is different we will utilize it's cfg
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if self.device_id == entry.data["id"]:
+                _LOGGER.info("Konnected host changed - creating replacement cfg entry")
+                config = copy.deepcopy(entry.data)
+                config[CONF_HOST] = self.host
+                config[CONF_PORT] = self.port
+                return await self.async_create_or_update_entry(config)
+
+        return await self.async_step_io()
+
+    async def async_step_io(self, user_input=None):
```

Comment:
Are any of the subsequent steps required to complete the connection to the panel? If they are just configuring optional items they should instead be part of a separate options flow, that the user can change after completing the main flow.

https://developers.home-assistant.io/docs/en/config_entries_options_flow_handler.html

Reply:
No.  I will migrate these to options flow.